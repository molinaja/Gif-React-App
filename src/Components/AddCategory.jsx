import { Component, useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ onNewCategory,  onErrorMsn}) => {

    const [inputValue, setInputValue] = useState('');  
    
    const onChangeInputValue = ({ target }) => {
        onErrorMsn('');
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();

        let errormsn = '';
        if (inputValue.trim().length === 0 ) {
            
            errormsn = "Debe haber minimo un caracter";
            onErrorMsn(errormsn);
        }else{

            onNewCategory(inputValue.trim());

        }
        
        setInputValue('');

    }

    return (
        <form onSubmit={(event) => onSubmit(event)}>
            <input
                id="Categoria"
                type="text"
                placeholder="Buscar Gif"
                value={inputValue}
                onChange={(event) => onChangeInputValue(event)}
            />
        </form>
    )
}

AddCategory.propTypes = {

    onNewCategory: PropTypes.func.isRequired,
    onErrorMsn: PropTypes.func.isRequired,


}

//export default AddCategory