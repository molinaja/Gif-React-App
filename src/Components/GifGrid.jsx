import React, { useEffect, useState } from 'react'
import {useFetchGifs} from '../Hooks/useFetchGifs'
import {GifGridItem} from './GifGridItem';

export const GifGrid = ({ category }) => {

    const{images, isloading} = useFetchGifs(category);  
   // console.log({images})
    return (
        <>
            <h3>
                {category}
            </h3>
            {
                isloading?(<h3>Cargando...</h3>):null
            }
            <div className='card-grid'>
                {
                    images.map(
                        (image) => (
                                <GifGridItem key={image.id} {...image}/>
                            )                    
                    )                
                }
            </div>

        </>
    )
}

//export default GifGrid