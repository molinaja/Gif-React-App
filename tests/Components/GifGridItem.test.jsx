import { render, screen } from "@testing-library/react";
import { GifGridItem } from '../../src/Components/GifGridItem';

describe('Prueba en <GifGridItem/>', () => {

    const title = 'gifTitulo';
    const url = 'www.test.com';

    test('Debe hacer match con el snapshot', () => {

        const { container } = render(<GifGridItem title={title} url={url} />);
        expect(container).toMatchSnapshot();    

    });

    test('Validando imagen cargada src y alt', ()=>{

        render(<GifGridItem title={title} url={url} />);
        const {src , alt} = screen.getByRole('img');
        expect( src ).toBe( 'http://localhost/' + url );
        expect( alt ).toBe( title );

    });

    test('Debe mostrar el titulo del gif', ()=>{

        render(<GifGridItem title = { title } url = { url } />);
        expect(screen.getByText( title ) ).toBeTruthy();

    });

});