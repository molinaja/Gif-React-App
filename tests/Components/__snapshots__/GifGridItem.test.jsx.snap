// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <GifGridItem/> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="gifTitulo"
      src="www.test.com"
    />
    <p>
      gifTitulo
    </p>
  </div>
</div>
`;
